### PR TITLE
MINIFICPP-1865 Integrate performance clang-tidy checks in CI

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-1865

-------------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: extensions/http-curl/tests/unit/InvokeHTTPTests.cpp [hunk 1]
```diff
@@ -238,14 +238,14 @@ TEST_CASE("HTTPTestsPostNoResourceClaim", "[httptest1]") {
   auto records = plan->getProvenanceRecords();
   std::shared_ptr<core::FlowFile> record = plan->getCurrentFlowFile();
   REQUIRE(record == nullptr);
-  REQUIRE(records.size() == 0);
+  REQUIRE(records.empty());
```
Comment (adamdebreceni) [inline extensions/http-curl/tests/unit/InvokeHTTPTests.cpp:241]: I feel like these transformations in `REQUIRE` are not advantageous, as in the former case we get an error like "error: 3 != 0", while in this new case we get "<...> is false"

Comment (lordgamez) [reply extensions/http-curl/tests/unit/InvokeHTTPTests.cpp:241]: I don't think we really care about the number of events in this test. The real question here is that if got any events or not. So if we break it and get any events here the count doesn't really help with the debugging IMO.

File: extensions/standard-processors/tests/integration/TestExecuteProcess.cpp [hunk 2]
```diff
@@ -87,17 +87,15 @@ int main(int /*argc*/, char ** /*argv*/) {
...
-  }
+  processor_workers.push_back(std::thread([processor, test_repo, &is_ready]() {
+    auto node = std::make_shared<core::ProcessorNode>(processor.get());
+    auto context = std::make_shared<core::ProcessContext>(node, nullptr, test_repo, test_repo);
+    context->setProperty(org::apache::nifi::minifi::processors::ExecuteProcess::Command, "sleep 0.5");
+    auto session = std::make_shared<core::ProcessSession>(context);
+    while (!is_ready.load(std::memory_order_relaxed)) {
+    }
+    processor->onTrigger(context.get(), session.get());
+  }));
```
Comment (adamdebreceni) [inline extensions/standard-processors/tests/integration/TestExecuteProcess.cpp:98]: I don't think these cases are interchangeable (I mean functionally sure they are), we should rework the whole test if we are sure that a single trigger is enough, I recon there was some idea behind testing multiple triggers, but for some reason was not materialized

File: libminifi/src/c2/C2Client.cpp [hunk 3]
```diff
@@ -177,7 +177,7 @@ std::shared_ptr<state::response::ResponseNode> C2Client::loadC2ResponseConfigura
             std::static_pointer_cast<state::response::ObjectNode>(prev_node)->add_node(new_node);
           }
         } else {
-          std::string optionName = option + "." + name;
+          std::string optionName = std::string(option).append(".").append(name);
```
Comment (adamdebreceni) [inline libminifi/src/c2/C2Client.cpp:180]: should we go with `StringUtils::join_pack` for such string concatenations?

File: extensions/standard-processors/tests/integration/TestExecuteProcess.cpp (hunk 2 above)
Comment (lordgamez) [reply extensions/standard-processors/tests/integration/TestExecuteProcess.cpp:98]: I agree it should be reworked and rewritten in the future, I added a jira ticket for this: https://issues.apache.org/jira/browse/MINIFICPP-1893

File: libminifi/src/c2/C2Client.cpp (hunk 3 above)
Comment (lordgamez) [reply libminifi/src/c2/C2Client.cpp]: Good idea, updated in 3b7a34d4fbb31fbb01acfa3654723ea05239f0fa

**Review by adamdebreceni — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

File: libminifi/test/unit/ExpectedTest.cpp [hunk 4]
```diff
@@ -52,7 +52,7 @@ TEST_CASE("expected map", "[expected][map]") {
 
   {
     const nonstd::expected<int, int> e = 21;
-    auto ret = std::move(e) | utils::map(mul2);
+    auto ret = e | utils::map(mul2);
```
Comment (szaszm) [inline libminifi/test/unit/ExpectedTest.cpp:55]: These are meant to test utilities on rvalues. You can see the same tests above on lvalues. Please revert these changes, except for the last one.

File: libminifi/test/unit/IdTests.cpp [hunk 5]
```diff
@@ -77,7 +77,7 @@ TEST_CASE("Test Generate Move", "[id]") {
 
   auto generated = generator->generate();
   auto str = generated.to_string();
-  utils::Identifier moved = std::move(generated);
+  utils::Identifier moved = generated;
```
Comment (szaszm) [inline libminifi/test/unit/IdTests.cpp:80]: This change is also removing an intentional move.

**Review by szaszm — CHANGES_REQUESTED** [review]

File: libminifi/src/c2/C2Agent.cpp [hunk 6]
```diff
@@ -739,7 +739,7 @@ utils::TaskRescheduleInfo C2Agent::produce() {
         std::make_move_iterator(payload_batch.end()),
         [&] (C2Payload&& payload) {
           try {
-            C2Payload && response = protocol_.load()->consumePayload(std::move(payload));
+            C2Payload && response = protocol_.load()->consumePayload(payload);
```
[the remark this answers is not in the document]
Comment (lordgamez) [reply libminifi/src/c2/C2Agent.cpp]: Updated in 7d86623adbe9d1cf018a62e72536bce62cd4332d

File: libminifi/test/unit/ExpectedTest.cpp (hunk 4 above)
Comment (lordgamez) [reply libminifi/test/unit/ExpectedTest.cpp]: Good catch, updated in 7d86623adbe9d1cf018a62e72536bce62cd4332d

File: libminifi/test/unit/IdTests.cpp (hunk 5 above)
Comment (lordgamez) [reply libminifi/test/unit/IdTests.cpp]: Updated in 7d86623adbe9d1cf018a62e72536bce62cd4332d

**Review by szaszm — APPROVED** [review]
